test/quantization/test_quantize_script.py

```diff
@@ -2348,10 +2339,81 @@ def forward(self, x):
                 return self.fc(x), size1, size2
 
         model = torch.jit.script(M()).eval()
-        data = torch.rand((1, 5), dtype=torch.float)
         qconfig_dict = {'': default_dynamic_qconfig}
 
-        model = quantize_dynamic_script(model, qconfig_dict, [data])
+        model = quantize_dynamic_script(model, qconfig_dict)
         FileCheck().check("quantized::linear_dynamic") \
                    .check_not("aten::_choose_qparams_per_tensor") \
                    .run(model.graph)
+
+    @override_qengines
+    def test_dynamic_shared_weights(self):
+        class myMod(torch.nn.Module):
+            def __init__(self, weight):
+                super().__init__()
+                self.linear = nn.Linear(5, 5)
+                self.linear.weight = weight
+
+            def forward(self, x):
+                return self.linear(x)
+
+        class DynamicModel(torch.nn.Module):
+            def __init__(self):
+                super(DynamicModel, self).__init__()
+                self.weight = torch.nn.Parameter(torch.ones(5, 5))
+                self.mod1 = myMod(self.weight)
+
+            def forward(self, x):
+                y = self.mod1(x)
+                z = torch.nn.functional.linear(y, self.weight)
+                return z
+
+        model = torch.jit.script(DynamicModel()).eval()
+        data = torch.randn(5, 5, dtype=torch.float)
+        for op in ['mod1', '']:
+            qconfig_dict = {op: default_dynamic_qconfig}
+            model_graph = quantize_dynamic_script(model, qconfig_dict)
+            out_graph = model_graph(data)
+
+            # Explicitly call forward on model before convert
+            m = prepare_dynamic_script(model, qconfig_dict)
+            m(data)
```

Comment:
Why is this needed?

Reply:
Also, in this test how are we verifying the final graph? We need to check if dynamic quantization occurs as expected.  We will need to clone weights and quantize only one of them  (i.e only the weights used by the module that has a qconfig_dict)

Reply:
This is how I make sure the numerics match between the two approaches, the `quantize_dynamic_script` call above doesn't take in the input data.